lib/session.js

```diff
@@ -108,6 +109,9 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 			if (user && (!passwordCheck || scmp(passwordCheck, hash(user.password)))) {
 				postHookedSigninWithUser(user, req, res, onSuccess, onFail);
 			} else {
+				if (!err) {
+					err = new Error('The password you entered is incorrect.');
```

Comment:
I updated the file to be more vague but I'll just say that on a lot of sites you're already exposed just by virtue of using the site, e.g. I already know what your username is on GitHub because you're posting messages to me. Github, Facebook, Twitter, et.al. leak information about their users everywhere (usernames appear on friends/connections lists, contributor lists, etc.) and they are still considered pretty secure as far as I know.